datashare-app/src/main/java/org/icij/datashare/web/UserResource.java

```diff
@@ -104,19 +106,24 @@ private static boolean parseBooleanQueryArg(String desc) {
         return parseBoolean(desc) || getStringValue(desc).isEmpty() || !desc.equalsIgnoreCase("false");
     }
 
-    @Operation(description = "Add event to history. The event's type, the project ids and the uri are passed in the request body.<br>" +
+    @Operation(description = "Add or update an event to user's history. The event's type, the project ids and the uri are passed in the request body.<br>" +
+            "To update the event's name, the eventId is required to retrieve the corresponding event." +
             "The project list related to the event is stored in database but is never queried (no filters on project)")
     @ApiResponse(responseCode = "200", description = "returns 200 when event is added or updated.")
     @Put("/me/history")
```

Comment:
I think there is some tech dept here: the creation should be done through a "POST" request. Therefore, the partial update should be done in a separated method with a "PATCH". This is non-blocking as it's not increasing the technical dept.

Reply:
Yes definitively something that can be improved with your suggestion, in recommendations PUT should be idempotent.